utils/PathSanitizingFileCheck

```diff
@@ -81,7 +81,7 @@ constants.""")
         # to escape it first, and then replace the escaped slash
         # literal (r'\\/') for our platform-dependent slash regex.
         stdin = re.sub(re.sub('\\\\/' if sys.version_info[0] < 3 else r'[/\\]',
-                              slashes_re, re.escape(pattern)),
+                              slashes_re, pattern),
```

Comment:
You cannot remove the `re.escape` or you will break cases like the one solved in #26024.

This has always worked with dashes in the folder names (mostly because most of my work directories are called `swift-source`). `re.escape` escapes those dashes, indeed, but since they are later used as a regular expression, it should match the literal dash correctly.

I think the incorrect pieces are the regular expressions used in the inner `re.sub`.

What's the problem you are seeing? Is there a way to reproduce? What are the original failures that led you to add the `r[/\\]` part? (stdin, replacement, and pattern)

Reply:
I'm seeing path substituion failures.  Its pretty easy to reproduce if you build in a path that contains `Windows-x86_64`.  The `-` gets quoted as: `'Windows\-x86_64` which then becomes `Windows(/|\\)-x86_64` which will fail to match.

Reply:
I suppose you are executing with Python 3, which means that `r'[/\\]'` is uses to search pieces. That regex means every "/" and every "\" will get replaced by the `slashes_re`, but you only want to replace "\\" (escaped Windows path separator) or "/".

The problem here (I just checked) is that Python 2 escapes "/" while Python 3 doesn't. In Python 3.8.1 I get `print(re.escape("/a/b/c/d/e")) #=> /a/b/c/d/e`, while in Python 2.7.1 `print(re.escape("/a/b/c/d/e")) #=> \/a\/b\/c\/d\/e`, which is not very helpful.

I think the solution is recovering the `re.escape`, and using `r'\\/' if sys.version_info[0] < 3 else r'/'` as the first parameter. That should cover the same cases as before, I think.

Reply:
I didn't get a notification for this, but ended up at the same place at the end after a bit of messing around with re in the interpreter.  Had I read this, it would have saved me a couple of hours.  Thanks though!